### PR TITLE
Add search to new build system

### PR DESCRIPTION
**Pull request by boxadesign**
### What is the context of this PR?
Add basic DS search to new build system.

### CODE REVIEW

**Comment by jrbarnes9** [comment]
[Search the design system](https://trello.com/c/gsFFJQCJ/66-search-the-design-system)

**Comment by jrbarnes9** [comment]
I don't think we need the 1px inset box-shadow on `.input:focus` for the search input.  It doesn't need it because it's already changing background colour on focus.

**Comment by jrbarnes9** [comment]
I think the search icon svg might need tweaking so the icon and label appear centred to the button boundary.  At the moment there's more space to the left of the icon than there is the right of the label.
Screenshot 2021-08-25 at 16 32 22

**Comment by jrbarnes9** [comment]
Is there any way we can add some sort of weighting to the title, so it ranks above the synonyms? 

Here for example, you'd expect search to be returned above autosuggest...
Screenshot 2021-08-31 at 12 02 17

**Comment by rmccar** [comment]
image
When the screen width goes down to 309px we get this issue

**Comment by boxadesign** [comment]
> image
> When the screen width goes down to 309px we get this issue

We don't suport anything below 320px

File: src/components/autosuggest/code.list.searcher.js [hunk 1]
```diff
@@ -6,7 +6,16 @@ export default function queryJson(query, data, searchFields) {
     threshold: 0.2,
     location: 1,
     distance: 1000,
-    keys: [searchFields, 'tags'],
+    keys: [
+      {
+        name: searchFields,
+        weight: 0.2,
```
Comment (rmccar) [inline src/components/autosuggest/code.list.searcher.js:12]: image
Found another place where weighting needs tweaking. Not tested fully but increasing the weight to 0.3 here solves this issue

Comment (boxadesign) [reply src/components/autosuggest/code.list.searcher.js:12]: I'll take a look. I thought that just having a greater weighting on one string query would always weight it higher 🤷🏼‍♂️

**Review by jrbarnes9 — CHANGES_REQUESTED** [review]
The footer overlaps the search results on short pages
Screenshot 2021-09-06 at 14 27 51

File: src/components/autosuggest/code.list.searcher.js (hunk 1 above)
Comment (rmccar) [reply src/components/autosuggest/code.list.searcher.js]: Yeah you would think that! It did fix the issue Jon found. I just made some quick changes and that fixed it. I can't quite remember exactly how the weighting works in Fuse.

**Review by rmccar — DISMISSED** [review]
Other than Jons footer issue, looks good to me

**Review by jrbarnes9 — APPROVED** [review]

**Review by rmccar — DISMISSED** [review]

**Review by LauraC119 — COMMENTED** [review]
Couple of synonyms it might be worth adding for the whole getting started area. I tried "Start" and "Set up" but neither had results. 

Also, not sure why the pages in Contribute and Getting started are categorised as "Documentation" in the search results. Shouldn't they be the same as components or patterns, which are categorised as "Components" or "Patterns"? Could confuse as there is no "Documentation" area.

**Review by kruncher — APPROVED** [review]

**Comment by jrbarnes9** [comment]
The yellow focus state border remains on after focus is moved from search input after something entered
Screenshot 2021-09-07 at 09 34 31

**Review by rmccar — APPROVED** [review]

**Review by jrbarnes9 — APPROVED** [review]

**Pull request merged**